dashboard/app/models/user.rb

```diff
@@ -305,6 +305,8 @@ class User < ApplicationRecord
     self.gender = Policies::Gender.normalize gender
   end
 
+  before_validation :student_in_lockout_flow?, on: :update
```

Comment:
Let's put the gates in their own method and change the name of the validation function to make its purpose, perhaps.

The name `student_in_lockout_flow?` seems to indicate that it is returning a boolean true/false and that it is informational rather than a validation.

```suggestion
  before_validation :enforce_age_or_state_update, on: :update, if: :should_check_age_or_state_update?
```

Reply:
If that feels like over-engineering, you can just introduce the block in the `if` via an anonymous function, but I feel like the logic is complicated enough to warrant a named function. But, to compare, it would be something like:

```
before_validation :enforce_age_or_state_update, on: :update, if: lambda do
  return false unless user.student?
  return false unless %w[US RD].include? country_code
  birthday_changed? || us_state_changed?
end
```